src/cargo/ops/lockfile.rs

```diff
@@ -66,6 +66,14 @@ pub fn write_pkg_lockfile(ws: &Workspace<'_>, resolve: &mut Resolve) -> CargoRes
         );
     }
 
+    if ws.is_locked() {
+        anyhow::bail!(
+            "Attempted to write to the standard library's lockfile.\n\
+            This most likely means the lockfile has been previously modified by mistake.\
+            Try removing and readding the `rust-src` component."
```

Comment:
* Lower case for first letter by convention in this repo.
* For rust-src component, it should be conditioned on the availability of rustup. You may use [`cargo::util::is_rustup`](https://github.com/rust-lang/cargo/blob/15fbd2f607d4defc87053b8b76bf5038f2483cf4/src/cargo/util/mod.rs?plain=1#L71) to check.

```suggestion
            "Attempted to write to the standard library's lockfile at: {}.\n\
            This most likely means the lockfile has been previously modified by mistake.\ 
            Try removing and readding the `rust-src` rustup component.
            lock_root.as_path_unlocked().join(LOCKFILE_NAME).display(),
```